main/inc/lib/api.lib.php

```diff
@@ -1276,9 +1278,78 @@ function api_protect_admin_script($allow_sessions_admins = false, $allow_drh = f
         return false;
     }
 
+    // see BT#17891
+    apiBlockInactiveUser();
+
     return true;
 }
 
+/**
+ * Function used to protect an inactive user.
+ *
+ * Function to determine an inactive user by the database
+ *
+ * @param int $user_id
+ *
+ * @return bool
```

Comment:
debería tener una descripción más amplia. Por ejemplo: `@return bool True if user is active (and it NOT blocked), false if the user is blocked.